zrml/prediction-markets/src/lib.rs

```diff
@@ -475,26 +473,66 @@ mod pallet {
                     T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?
                 }
             }
-            Self::remove_last_dispute_from_market_ids_per_dispute_block(&disputes, &market_id)?;
```

Comment:
Good observation! I got that in my mind while doing the modification. Because the MDMs now control the resolution, I moved that code to the MDMs. In this case [here](https://github.com/zeitgeistpm/zeitgeist/blob/818c68a8435e57ca5af5de88a59e81143f4335a0/zrml/simple-disputes/src/lib.rs#L143-L148).

In case of `authorized`, the resolution happens after the correction period [here](https://github.com/zeitgeistpm/zeitgeist/blob/818c68a8435e57ca5af5de88a59e81143f4335a0/zrml/authorized/src/lib.rs#L84-L90).

For `court` we need to keep this in mind.

Reply:
Okay, fair enough. But this means that `prediction-markets` doesn't clean up the auto-resolve that stems from the oracle report. Looking at this in terms of responsibility, why is the MDM responsible for clearing up an auto-resolve that `prediction-markets` added. And looking at this in practical terms, the following test will pass if the MDM doesn't clear up the auto-resolve from the report (as is the case for `simple-disputes`):

```rust
let end = 10;
simple_create_categorical_market(MarketCreation::Permissionless, 0..end, ScoringRule::CPMM);
let market = MarketCommons::market(&0).unwrap();
run_to_block(end + market.deadlines.grace_period);
assert_ok!(PredictionMarkets::report(
    Origin::signed(BOB),
    0,
    OutcomeReport::Categorical(1)
));
run_blocks(market.deadlines.dispute_duration - 1);
assert_ok!(PredictionMarkets::dispute(
    Origin::signed(CHARLIE),
    0,
    OutcomeReport::Categorical(0)
));
run_blocks(2);
let market = MarketCommons::market(&0).unwrap();
assert_eq!(market.status, MarketStatus::Resolved);
assert_eq!(market.report.unwrap().outcome, OutcomeReport::Categorical(1));
```

I can't think of any MDM which would want this type of behavior. I suggest that each component is responsible of cleaning up every auto-resolve they pushed. So when `prediction-markets` transfers the responsiblity for resolving the market to the MDM, it cleans up its own auto-resolves and lets the MDM set new ones.

Reply:
> Okay, fair enough. But this means that prediction-markets doesn't clean up the auto-resolve that stems from the oracle report.

Auto-resolve for `MarketIdsPerReportBlock` is cleaned up [here](https://github.com/zeitgeistpm/zeitgeist/blob/a9cacb3195bc159c0a2f40f56b767ea9e3753c7e/zrml/prediction-markets/src/lib.rs#L2685). 

> Looking at this in terms of responsibility, why is the MDM responsible for clearing up an auto-resolve that prediction-markets added.

Items of `MarketIdsPerDisputeBlock` are only added per `add_auto_resolve`, which is called by the MDM. Could be cleared by `prediction-markets` for example for `destroy_market`, but should also be clearable by the MDM (like simple-disputes is a good example for this). 

Items of `MarketIdsPerReportBlock` are only added by `fn report`. Removed by this [call](https://github.com/zeitgeistpm/zeitgeist/blob/a9cacb3195bc159c0a2f40f56b767ea9e3753c7e/zrml/prediction-markets/src/lib.rs#L2685).

> And looking at this in practical terms, the following test will pass if the MDM doesn't clear up the auto-resolve from the report

So, what is the problem of the test here? The mock is configured like that, right? After a dispute, simple-disputes extends the dispute period again.

> I suggest that each component is responsible of cleaning up every auto-resolve they pushed.

This is what I currently do [here](https://github.com/zeitgeistpm/zeitgeist/blob/7f6bc000176c10e6bd4fb15a17aa4a21445138c7/zrml/simple-disputes/src/lib.rs#L145).

> Why force every single MDM to remove the auto-resolve of the report?

The auto-resolution is known by the MDM. We only like to have one auto-resolution. Yes, the MDM is currently responsible of keeping track of its auto-resolution.